### PR TITLE
chore(agw): upload journal log to sentry

### PR DESCRIPTION
**Pull request by ajahl**
Signed-off-by: Alex Jahl <alexander.jahl@tngtech.com>

## Summary

The PR is a prototype for uploading journal log information to Sentry when a c service crashes.

Approach: We used the `before_send` callback method that is called by the Sentry SDK before a crash report is sent. The callback extracts certain information from `journalctl` log and provide it stored in a separate log file,  [magma_services.log](https://github.com/magma/magma/files/8124561/magma_services.log). The crash report contains the log file as attachment. 

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: orc8r/gateway/c/common/sentry/SentryWrapper.cpp [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #define SHARED_MCONFIG "shared_mconfig"
 #define SHOULD_UPLOAD_MME_LOG "sentry_upload_mme_log"
 #define MME_LOG_PATH "/var/log/mme.log"
+#define SYS_LOG_PATH "/var/log/magma_services.log"
```
Comment (alexzurbonsen) [inline orc8r/gateway/c/common/sentry/SentryWrapper.cpp]: While reading this again: should we rename `SYS_LOG_PATH` to something that reveals that we are writing to this path and not reading from it? The current name might suggest that it is the path to the syslog file, analogous to  `MME_LOG_PATH`.

Comment (ajahl) [reply orc8r/gateway/c/common/sentry/SentryWrapper.cpp]: We write the file first and afterwards read it via Sentry SDK. I renamed in `JOURNAL_LOG_PATH`.

**Review by alexzurbonsen — APPROVED** [review]

**Comment by Neudrino** [comment]
Thanks @themarwhal and @ardzoht for the insights shared today!

There were two items which I suggest to address separately in respective follow up PRs:
- providing SCTPd logs, and
- making the number of lines configurable via streamed mconfig.

File: orc8r/gateway/c/common/sentry/SentryWrapper.cpp [hunk 2]
```diff
@@ -115,6 +116,14 @@ std::string get_snowflake() {
   return buffer.str();
 }
 
+static sentry_value_t before_send(sentry_value_t event, void* hint,
+                                  void* data) {
+  std::string command = "journalctl -u magma@* -n 1000 > ";
```
Comment (sebathomas) [inline orc8r/gateway/c/common/sentry/SentryWrapper.cpp]: To get sctpd as well:
```suggestion
  std::string command = "journalctl -u sctpd -u magma@* -n 1000 > ";
```

**Review by Neudrino — APPROVED** [review]

File: orc8r/cloud/go/services/orchestrator/obsidian/models/swagger.v1.yml [hunk 3]
```diff
@@ -1451,6 +1451,11 @@ definitions:
       upload_mme_log:
         type: boolean
         example: false
+      number_of_lines_in_log:
+        type: number
+        format: uint32
+        minimum: 0
```
Comment (hcgatewood) [inline orc8r/cloud/go/services/orchestrator/obsidian/models/swagger.v1.yml]: This might be unnecessary given the object's type (?)

**Review by hcgatewood — APPROVED** [review]
Cloud changes lgtm

**Review by themarwhal — APPROVED** [review]
Woohoo!

**Review by alexzurbonsen — CHANGES_REQUESTED** [review]
We should also modify the go code that translates from swagger to proto, such that it reflects the changes.

**Comment by sebathomas** [comment]
Right. I think that's:
* lte/cloud/go/services/lte/servicers/builder_servicer.go
* orc8r/cloud/go/services/orchestrator/servicers/builder_servicer.go

**Comment by ajahl** [comment]
> We should also modify the go code that translates from swagger to proto, such that it reflects the changes.


> Right. I think that's:
> 
> * lte/cloud/go/services/lte/servicers/builder_servicer.go
> * orc8r/cloud/go/services/orchestrator/servicers/builder_servicer.go

It's done. I modified the go files.

**Review by sebathomas — APPROVED** [review]
Tried it on my machine and confirmed that the number of lines is forwarded to the mconfig file in the AGW.

**Review by alexzurbonsen — APPROVED** [review]

**Comment by sebathomas** [comment]
Just noticed the `yarn test` error in the CI. Rerunning to see if it's flaky.

**Review by sebathomas — CHANGES_REQUESTED** [review]
Nope, I think it's consistently red. We should fix that before merging.

**Review by sebathomas — APPROVED** [review]
This was not broken by the PR, it's also red on master. Opened another ticket to track that:
#11895

**Pull request merged**